src/client/client.ts

```diff
@@ -110,38 +110,35 @@ export class SubscriptionClient {
     this.client.close();
   }
 
-  public subscribe(options: SubscriptionOptions, handler: (error: Error[], result?: any) => void) {
-    const { query, variables, operationName, context } = options;
-
-    if (!query) {
-      throw new Error('Must provide `query` to subscribe.');
-    }
+  public query(options: RequestOptions): Promise<ExecutionResult> {
+    return new Promise((resolve, reject) => {
+      const handler = (error: Error[], result?: any) => {
+        if (result) {
+          resolve({
```

Comment:
this.executeRequest should return reqId which you will need to call some teardown function for it before resolving the promise.